pull-versions.js

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env node
+
+const util = require('util')
+const fs = require('fs')
+const rimraf = util.promisify(require('rimraf'))
+const Git = require('nodegit')
+
+const writeFile = util.promisify(fs.writeFile)
+
+const baselineRef = process.argv[2]
+const comparisonRef = process.argv[3]
+
+;(async () => {
+  if (!baselineRef || !comparisonRef) {
+    throw new Error('Expected to receive two ref args. Cannot continue.')
+  }
+
+  await rimraf('./underscore')
+  const repo = await Git.Clone('git://github.com/jashkenas/underscore.git', './underscore')
+
+  const checkouts = [{ name: 'baseline', ref: baselineRef }, { name: 'comparison', ref: comparisonRef }]
+  for (const { name, ref } of checkouts) {
+    let commit
+    try {
+      commit = await repo.getReferenceCommit(ref)
```

Comment:
I am thoroughly confused about how to checkout a ref using `nodegit`. There seem to be different commands for all sorts of things, but there does not seem to be an equivalent of `git checkout`.

I need to revisit this when I find the time.